src/librustc_lint/types.rs

```diff
@@ -636,11 +636,12 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
                         if all_phantom { FfiPhantom(ty) } else { FfiSafe }
                     }
                     AdtKind::Union => {
-                        if !def.repr.c() {
+                        if !def.repr.c() && !def.repr.transparent() {
                             return FfiUnsafe {
                                 ty: ty,
                                 reason: "this union has unspecified layout",
-                                help: Some("consider adding a #[repr(C)] attribute to this union"),
+                                help: Some("consider adding a #[repr(C)] or #[repr(transparent)] \
```

Comment:
```suggestion
                                help: Some("consider adding a `#[repr(C)]` or `#[repr(transparent)]` \
```

Reply:
Could you also update the message for structs to add the backticks for conformity?